src/tarantool_schema.c

```diff
@@ -325,12 +349,75 @@ decode_index_parts_166(struct schema_field_value *parts, uint32_t part_count,
 		uint32_t len;
 		const char *str = mp_decode_str(data, &len);
 		part->field_type = field_type_by_name(str, len);
+		if (part->field_type == field_type_MAX)
+			return -1;
 
 		for (uint32_t j = 2; j < item_count; ++j)
 			mp_next(data);
-		/* Set default values. */
-		part->is_nullable = false;
-		part->coll_id = COLL_NONE;
+	}
+	return 0;
+}
+
+static int
+decode_index_part(struct schema_field_value *part, uint32_t map_size,
+		  const char **data)
+{
+	*part = field_val_def;
+	for (uint32_t i = 0; i < map_size; ++i) {
+		if (mp_typeof(**data) != MP_STR)
+			return -1;
+
+		uint32_t k_len;
+		const char *key = mp_decode_str(data, &k_len);
+		if (strncmp_strict(key, k_len, "type")) {
+			if (mp_typeof(**data) != MP_STR)
+				return -1;
+			uint32_t v_len;
+			const char *val = mp_decode_str(data, &v_len);
+			part->field_type = field_type_by_name(val, v_len);
+			if (part->field_type == field_type_MAX)
+				return -1;
```

Comment:
The same question as for `decode_index_parts_166()`.

Reply:
Fixed

Reply:
Continued the discussion in https://github.com/tarantool/tarantool-php/pull/164#discussion_r454719648

Reply:
I have implemented 2.3 from https://github.com/tarantool/tarantool-php/pull/164#discussion_r456708834